mlfinlab/portfolio_optimization/hcaa.py

```diff
@@ -52,19 +52,21 @@ def _compute_cluster_inertia(labels, asset_returns):
     def _get_optimal_number_of_clusters(self,
                                         correlation,
                                         asset_returns,
+                                        linkage,
                                         num_reference_datasets=5,
                                         max_number_of_clusters=10):
         """
         Find the optimal number of clusters for hierarchical clustering using the Gap statistic.
 
         :param correlation: (np.array) matrix of asset correlations
         :param asset_returns: (pd.DataFrame) historical asset returns
+        :param linkage: (str) the type of linkage method to use for clustering
```

Comment:
You don't specify the types of strings it can take? Please include them.

Reply:
@Jackal08 I have specified them in the main allocate method which is a public one. Since these are private methods I did not specify